cucumber/authenticators_azure/features/authn_azure.feature

```diff
@@ -0,0 +1,49 @@
+Feature: Azure Authenticator - Users can authenticate with Azure authenticator
+
+  Background:
+    Given a policy:
+    """
+    - !policy
+      id: conjur/authn-azure/prod
+      body:
+      - !webservice
+
+      - !variable
+        id: provider-uri
+
+      - !group
+        id: apps
+
+      - !permit
+        role: !group apps
+        privilege: [ read, authenticate ]
+        resource: !webservice
+    """
+    And I am the super-user
+    And I successfully set Azure variables
+    And I have host "no-assigned-identity-app"
+    And I set Azure annotations to host "no-assigned-identity-app"
+    And I grant group "conjur/authn-azure/prod/apps" to host "no-assigned-identity-app"
+
+  Scenario: A valid Azure token to get Conjur access token
+    # We want to verify the returned access token is valid for retrieving a secret
+    Given I have a "variable" resource called "test-variable"
+    And I permit host "no-assigned-identity-app" to "execute" it
+    And I add the secret value "test-secret" to the resource "cucumber:variable:test-variable"
+
+    And I fetch an Azure access token inside machine
+    When I authenticate via Azure with token as host "no-assigned-identity-app"
+    Then host "no-assigned-identity-app" is authorized
```

Comment:
authorized to do what? authorized to authenticate?

Reply:
"has been authorized by Conjur" or just "has been authorized". By this message, it isn't clear that this means it has been given an access token by Conjur

Reply:
authorized by Conjur - got a 200 ok response. This is not a new step so if you think it should be changed, let's do it in a separate PR to master

Reply:
I think it is unclear. Only I looked at the code, it became clear. IMO worth the refactor

Reply:
Created PR - https://github.com/cyberark/conjur/pull/1425